fairtools/MCConfigurator/FairXmlVMCConfig.h

```diff
@@ -0,0 +1,80 @@
+/********************************************************************************
+ *    Copyright (C) 2022 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH    *
+ *                                                                              *
+ *              This software is distributed under the terms of the             *
+ *              GNU Lesser General Public Licence (LGPL) version 3,             *
+ *                  copied verbatim in the file "LICENSE"                       *
+ ********************************************************************************/
+// -------------------------------------------------------------------------
+// -----               FairXmlVMCConfig header file                    -----
+// -----            Created 2022.10.04 by R. Karabowicz                -----
+// -------------------------------------------------------------------------
+
+#ifndef FAIR_XML_VMC_CONFIG_H
+#define FAIR_XML_VMC_CONFIG_H
+
+#include "FairGenericVMCConfig.h"
+#include "FairLogger.h"
+#include "TXMLEngine.h"
+
+#include <string>
+
+using namespace std;
+
+class FairXmlVMCConfig : public FairGenericVMCConfig
+{
+  public:
+    FairXmlVMCConfig();
+    virtual ~FairXmlVMCConfig() {}
+
+    virtual void Setup(const char* mcEngine);
+    virtual void SetupPostInit(const char* mcEngine);
+
+    virtual void UsePostInitConfig(bool useC = true, const char* stringC = "g4ConfigPostInit.xml")
+    {
+        fPostInitFlag = useC;
+        fPostInitName = stringC;
+    }
+
+  private:
+    string ObtainXmlFileName(const char* mcEngine);
+    void StoreXmlInfo();
+
+    string fMCEngine;
+
+  protected:
+    void SetupGeant3();
+    void SetupGeant4();
+    virtual void SetupStack() = 0;
+    void SetCuts();
+
+    TXMLEngine* fXmlEngine;
```

Comment:
This object is leaking ...

No need for a heap object here anyways, let's keep it a plain simple value member.

```suggestion
    TXMLEngine fXmlEngine;
```